packages/yourloops/pages/signup/signup-account-form.tsx

```diff
@@ -0,0 +1,254 @@
+import IconButton from "@material-ui/core/IconButton";
+import Visibility from "@material-ui/icons/Visibility";
+import VisibilityOff from "@material-ui/icons/VisibilityOff";
+import Button from "@material-ui/core/Button";
+
+import { useSignUpFormState } from "./signup-formstate-context";
+import { REGEX_EMAIL } from "../../lib/utils";
+import appConfig from "../../lib/config";
+import SignUpFormProps from "./signup-form-props";
+
+interface Errors {
+  userName: boolean;
+  newPassword: boolean;
+  confirmNewPassword: boolean;
+}
+
+const formStyle = makeStyles((theme: Theme) => {
+  return {
+    Button: {
+      marginRight: theme.spacing(1),
+    },
+    TextField: {
+      marginLeft: theme.spacing(0),
+      marginRight: theme.spacing(1),
+    },
+    Checkbox: {
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+    },
+  };
+});
+
+/**
+ * SignUpAccount Form
+ */
+function SignUpAccountForm(props: SignUpFormProps): JSX.Element {
+  const { t } = useTranslation("yourloops");
+  const { state, dispatch } = useSignUpFormState();
+  const { handleBack, handleNext } = props;
+  const defaultErr = {
+    userName: false,
+    newPassword: false,
+    confirmNewPassword: false,
+  };
+  const [errors, setErrors] = React.useState<Errors>(defaultErr);
+  const [newPassword, setNewPassword] = useState("");
+  const [showNewPassword, setShowNewPassword] = useState(false);
+  const [showConfirmNewPassword, setShowConfirmNewPassword] = useState(false);
+  const classes = formStyle();
+
+  const onChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+    keyField: string,
+    setState?: React.Dispatch<React.SetStateAction<string>>
+  ): void => {
+    if (!setState) {
+      dispatch({
+        type: "EDIT_FORMVALUE",
+        key: keyField,
+        value: event.target.value,
+      });
+    } else {
+      setState(event.target.value);
+    }
+  };
+
+  const onClick = (
+    _event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    showPassword: boolean,
+    setState: React.Dispatch<React.SetStateAction<boolean>>
+  ): void => {
+    if (showPassword) {
+      setState(false);
+    } else {
+      setState(true);
+    }
+  };
+
+  const isUserNameValid = (): boolean => {
+    const err =
+      _.isEmpty(state.formValues?.accountUsername.trim()) ||
+        !REGEX_EMAIL.test(state.formValues?.accountUsername);
+
+    setErrors({ ...errors, userName: err });
+    return !err;
+  };
+
+  const isPasswordValid = (): boolean => {
+    const err =
+      _.isEmpty(newPassword?.trim()) ||
+      newPassword?.length < appConfig.PASSWORD_MIN_LENGTH;
+    setErrors({ ...errors, newPassword: err });
+    return !err;
+  };
+
+  const isConfirmNewPasswordValid = (): boolean => {
+    const err =
+      _.isEmpty(state.formValues?.accountPassword.trim()) ||
+      state.formValues?.accountPassword !== newPassword;
+    setErrors({ ...errors, confirmNewPassword: err });
+    return !err;
+  };

```

Comment:
The name of these functions makes me think that they are just simple getters, but since they act on the state, maybe it should be changed?

Reply:
indeed, it should be split into two functions